### DISCUSSION TITLE
Nested repeaters returning as blocks

### DISCUSSION THREAD

**Discussion by ghost** [area17/twill#1219] [Q&A] [answered]
Hi,

As the title says, when I try to nest repeaters, all the repeaters after the top level are returned as blocks, which is causing issues when saving the lower level repeaters. I've seen mention of using JsonRepeaters but they aren't fit for purpose for what I'm trying to do:

![image](https://user-images.githubusercontent.com/80678577/139241105-f5b347ef-3117-4d46-8ddc-77b61078d5b7.png)

In the screen above, we have 3 object types at play, each with their own table/model/repo/etc: Quiz, QuizQuestion, and QuizQuestionOption
A Quiz can have many QuizQuestions, and a QuizQuestion can have many QuizQuestionOptions. I'd be looking to store each new entity as it's own object in the DB in their respective tables, not as json anywhere.

Is there any way to convert the blocks to repeaters or to have them return as repeaters in the first place?

**Comment by pboivin** [comment, marked as answer]
Hi @aarontilt,

That's right, nested repeaters are not supported outside of the block editor at the moment. You may have found [this thread](https://github.com/area17/twill/issues/1060#issuecomment-888062137) already, which has a hack to save nested repeaters as JSON, but I don't think I would recommend this solution for anything other than exploring Twill internals.

This will definitely be addressed in the near future but for now, it's probably best to refactor to a block editor if possible.

**Reply by ghost** [reply]
fair enough, thank you for the response!

**Reply by ifox** [reply]
Another way to deal with this is with parent- child modules using dot notation, as mentioned in #508.

**Reply by ifox** [reply]
You would have an index of quizzes and another index view for each quizz, that would list quizz questions. On the quizz question form you would be able to use repeaters for your question options.
